src/acl/agent.ts

```diff
@@ -147,6 +155,57 @@ export function unstable_getAgentResourceAccessModesAll(
   return getAccessModesByAgent(agentResourceRules);
 }
 
+/**
+ * Given an ACL LitDataset, modify the ACL Rules to set specific Access Modes for a given Agent.
+ *
+ * If the given ACL LitDataset already includes ACL Rules that grant a certain set of Access Modes
+ * to the given Agent, those will be overridden by the given Access Modes.
+ *
+ * Keep in mind that this function will not modify:
+ * - access arbitrary Agents might have been given through other ACL rules, e.g. public or group-specific permissions.
+ * - what access arbitrary Agents have to child Resources.
+ *
+ * Also, please note that this function is still experimental: its API can change in non-major releases.
+ *
+ * @param aclDataset The LitDataset that contains Access-Control List rules.
+ * @param agent The Agent to grant specific Access Modes.
+ * @param accessModes The Access Modes to grant to the Agent.
+ */
+export function unstable_setAgentResourceAccessModes(
+  aclDataset: unstable_AclDataset,
+  agent: WebId,
+  accessModes: unstable_AccessModes
```

Comment:
Yes, I was operating on that assumption. I guessed that the vast, vast majority use case for the foreseeable future would be that each Resource has its own distinct ACL file, and that bringing in the ability to modify rules applying to different Resources would be confusing enough to warrant having those who do want to manipulate rules for other Resources fetch the same ACL again. That sounded reasonable to me, but happy to discuss of course :)

Reply:
I agree that in most cases, your assumption will hold, so I'm okay to go with it. However, I am still unsure if modifying resource-scoped ACL in a document that isn't directly linked to the target resource (let's say its parent's ACL document) is in-standard, but if so I think I'd prefer to have a `unstable_setAgentResourceAccessModesForResource`, that does not use the `aclDataset.accessTo` parameter, because otherwise we'll make it harder to work with servers having different ways of representing ACLs. 

However, since it would be a non-breaking change to add it later, I'm fine going with the current approach now and see if the need for the alternative arises in the future.

Reply:
Hmm... But _if_ it is in-spec, then we should probably remove the `accessTo` property altogether, since its only purpose is to indicate what the assumed subject of the ACL is. And that _would_ be a breaking change, so let's wait until we hear back from the spec people.

Reply:
Here too: apparently [it is invalid for a Rule to point to a different Resource](https://github.com/solid/specification/issues/55#issuecomment-536092174).

Reply:
That settles it then.